src/Sylius/Bundle/CoreBundle/Doctrine/ORM/Handler/ResourceDeleteHandler.php

```diff
@@ -24,19 +25,29 @@ final class ResourceDeleteHandler implements ResourceDeleteHandlerInterface
     /** @var ResourceDeleteHandlerInterface */
     private $decoratedHandler;
 
-    public function __construct(ResourceDeleteHandlerInterface $decoratedHandler)
+    /** @var EntityManagerInterface */
+    private $entityManager;
+
+    public function __construct(ResourceDeleteHandlerInterface $decoratedHandler, EntityManagerInterface $entityManager)
     {
         $this->decoratedHandler = $decoratedHandler;
+        $this->entityManager = $entityManager;
     }
 
     /**
      * @throws DeleteHandlingException
      */
     public function handle(ResourceInterface $resource, RepositoryInterface $repository): void
     {
+        $this->entityManager->beginTransaction();
+
         try {
             $this->decoratedHandler->handle($resource, $repository);
+
+            $this->entityManager->commit();
         } catch (ORMException $exception) {
+            $this->entityManager->rollback();
+
             throw new DeleteHandlingException();
```

Comment:
It's always good to keep (if possible) previous exception included to get stacktrace if needed.